### ISSUE TITLE
de-capitalizing headers produces server errors with some backends

### ISSUE THREAD

**Issue by bradgearon** [greim/hoxy#28]
suggestion is to use title case or not modify headers that have been set
does this seem plausible?

**Comment by sholladay** [comment]
Node 0.12.0 is out and support for this is now possible, for anyone willing to implement it.
https://nodejs.org/api/http.html#http_message_rawheaders

When doing so, I would personally appreciate making it configurable, since this is spec breaking behavior.

**Comment by greim** [comment]
@sholladay - As long as `headerReceived.toLowerCase() === headerSent.toLowerCase()` would it be spec-breaking? I probably need to read up on the nuances but was under the impression headers were case insensitive, e.g. `CoNtEnT-TyPe` would be spec-valid (although not probably handled well by some servers in the wild).

**Comment by sholladay** [comment]
Correct. What I mean to say is that needing this as a feature is effectively spec-breaking, as it shows you are failing to NOT care about the casing of headers. But I do like choices and for a proxy in particular, it makes sense to be as transparent as possible unless otherwise desired.
